addons/mail/static/src/js/composer.js

```diff
@@ -469,7 +469,16 @@ var BasicComposer = Widget.extend(chat_mixin, {
     preprocess_message: function () {
         // Return a deferred as this function is extended with asynchronous
         // behavior for the chatter composer
-        var value = _.escape(this.$input.val()).replace(/\n|\r/g, '<br/>');
+
+        //Remove Extra newlines from starting and ending of Message.
+        var value = _.escape(this.$input.val()).replace(/^(\r|\n)+|(\n|\r)+$/g, '');
+
+        //Combine 2 or more consecutive newlines into 1 newline.
+        value = value.replace(/(\r|\n){2,}/g, '<br/><br/>');
+
+        //convert newline to <br/> tag.
+        value = value.replace(/(\r|\n){1,}/g, '<br/>');
```

Comment:
I understand what this is used for, what I mean is that line 480 currently has modifier {1,} which is equivalent to + and will just convert any sequence of at least 1 occurence, but sequences of at least *2* occurrences are already converted line 477, so `{1,}` does not seem very useful since it should only ever match `{1}` which is the default (and same as not putting an occurrence constraint).